### PR TITLE
👩‍🔬 [Investment Day] Swiftlint Cleanup

### PR DESCRIPTION
**Pull request by dusi**
# 📲 What

* Removes unused rules
* Prefers to use `next` where possible

# 🔨 How

Where possible I've replaced `// swiftlint:disable` with `// swiftlint:disable:next` or disable block. If a specific file disabled a rule for the whole file (at the beginning of the file), I would have replaced this with `next` only if the rule wasn't broken more than 5 times. If it was broken more times it was an indicator that the file was assuming / working differently than we would like which is a work for some other time.

# 🤔 Why

Best practices

# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| 0 Swiftlint warnings after `Clean Build` | 0 Swiftlint warnings after `Clean Build`  |

# ✅ Acceptance criteria

- [x] All tests pass
- [ ] No `Swiftlint` warnings

### CODE REVIEW

File: Kickstarter-iOS/AppDelegate.swift [hunk 1]
```diff
@@ -191,7 +191,7 @@ internal final class AppDelegate: UIResponder, UIApplicationDelegate {
       .observeForUI()
       .observeValues { [weak self] in self?.findRedirectUrl($0) }
 
-    //swiftlint:disable discarded_notification_center_observer
+    // swiftlint:disable discarded_notification_center_observer
```
Comment (dusi) [inline Kickstarter-iOS/AppDelegate.swift]: For consistency with the rest of the codebase `// swiftlint` is preferred to `//swiftlint`

File: Kickstarter-iOS/DataSources/SettingsAccountDataSourceTests.swift [hunk 2]
```diff
@@ -41,10 +40,18 @@ final class SettingsAccountDataSourceTests: XCTestCase {
                                   shouldHideEmailWarning: true,
                                   shouldHideEmailPasswordSection: false)
 
-    XCTAssertEqual(SettingsAccountCellType.changeEmail, self.dataSource.cellTypeForIndexPath(indexPath: indexPath1))
-    XCTAssertEqual(SettingsAccountCellType.changePassword, self.dataSource.cellTypeForIndexPath(indexPath: indexPath2))
-    XCTAssertEqual(SettingsAccountCellType.privacy, self.dataSource.cellTypeForIndexPath(indexPath: indexPath3))
-    XCTAssertEqual(SettingsAccountCellType.paymentMethods, self.dataSource.cellTypeForIndexPath(indexPath: indexPath4))
+    XCTAssertEqual(
```
Comment (dusi) [inline Kickstarter-iOS/DataSources/SettingsAccountDataSourceTests.swift]: Splitting up the lines didn't make the readability worse and allowed us to get rid of the rule altogether.

File: KsApi/MockService.swift [hunk 3]
```diff
@@ -1455,7 +1455,6 @@ private extension MockService {
       }
     )
   }
-  // swiftlint:enable type_name
```
Comment (dusi) [inline KsApi/MockService.swift]: Unused rule...there were couple others in this PR.

File: Library/ViewModels/PaymentMethodsViewModel.swift [hunk 4]
```diff
@@ -62,11 +62,7 @@ PaymentMethodsViewModelInputs, PaymentMethodsViewModelOutputs {
     self.showAlert = deletePaymentMethodEventsErrors
       .ignoreValues()
       .map {
-        localizedString(
-          key: "Something_went_wrong_and_we_were_unable_to_remove_your_payment_method_please_try_again",
-          //swiftlint:disable:next line_length
-          defaultValue: "Something went wrong and we were unable to remove your payment method, please try again."
-        )
+        Strings.Something_went_wrong_and_we_were_unable_to_remove_your_payment_method_please_try_again()
```
Comment (dusi) [inline Library/ViewModels/PaymentMethodsViewModel.swift]: Looks like localized string was available but we didn't use it 🤷‍♂️

Comment (Scollaco) [reply Library/ViewModels/PaymentMethodsViewModel.swift]: Actually it was being used. The `localizedString(key:defaultValue:)` function is exactly what `Something_went_wrong...` string function returns. If I'm not wrong, this was implemented that way so we could merge the PR before having the strings translated. But good catch though. 😄

**Pull request merged**